File: Python/769.最多能完成排序的块.py
from typing import List
import sys

class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ans: int = 0
        maxn = -sys.maxsize - 1
        for i in range(len(arr)):
            maxn = max(maxn, arr[i])
            if i == maxn:
                ans += 1
        return ans

File: Golang/769.最多能完成排序的块.go
package leetcode

func maxChunksToSorted(arr []int) int {
	ans := 0
	maxn := -1 << 31
	for k, v := range arr {
		maxn = maxInt(maxn, v)
		if k == maxn {
			ans++
		}
	}
	return ans
}

func maxInt(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
